RecoBTag/ONNXRuntime/plugins/ParticleTransformerAK4SonicJetTagsProducer.cc

```diff
@@ -0,0 +1,306 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Framework/interface/makeRefToBaseProdFrom.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/BTauReco/interface/JetTag.h"
+
+#include "DataFormats/BTauReco/interface/ParticleTransformerAK4TagInfo.h"
+#include "DataFormats/BTauReco/interface/ParticleTransformerAK4Features.h"
+
+#include "HeterogeneousCore/SonicTriton/interface/TritonEDProducer.h"
+
+#include "HeterogeneousCore/SonicTriton/interface/TritonData.h"
+
+class ParticleTransformerAK4SonicJetTagsProducer : public TritonEDProducer<> {
+public:
+  explicit ParticleTransformerAK4SonicJetTagsProducer(const edm::ParameterSet&);
+  ~ParticleTransformerAK4SonicJetTagsProducer() override;
+
+  void acquire(edm::Event const &iEvent, edm::EventSetup const &iSetup, Input &iInput) override;
+
+  void produce(edm::Event &iEvent, edm::EventSetup const &iSetup, Output const &iOutput) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  typedef std::vector<reco::ParticleTransformerAK4TagInfo> TagInfoCollection;
+  typedef reco::JetTagCollection JetTagCollection;
+
+  const edm::EDGetTokenT<TagInfoCollection> src_;
+  std::vector<std::string> flav_names_;
+  std::vector<std::string> input_names_;
+  std::vector<std::string> output_names_;
+
+  enum InputIndexes {
+    kChargedCandidates = 0,
+    kNeutralCandidates = 1,
+    kVertices = 2,
+    kChargedCandidates4Vec = 3,
+    kNeutralCandidates4Vec = 4,
+    kVertices4Vec = 5
+  };
+  
+  constexpr static unsigned n_features_cpf_ = 16;
+  constexpr static unsigned n_pairwise_features_cpf_ = 4;
+  constexpr static unsigned n_features_npf_ = 8;
+  constexpr static unsigned n_pairwise_features_npf_ = 4;
+  constexpr static unsigned n_features_sv_ = 14;
+  constexpr static unsigned n_pairwise_features_sv_ = 4;
```

Comment:
don't copy/reimplement constants. move the ones from https://github.com/cms-sw/cmssw/blob/master/RecoBTag/ONNXRuntime/plugins/ParticleTransformerAK4ONNXJetTagsProducer.cc into a helper class or base class.

Reply:
move to /interface/tensor_config.h